errai-security/errai-security-server/src/main/java/org/jboss/errai/security/server/servlet/UserHostPageFilter.java

```diff
@@ -98,13 +98,26 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
       else {
         output = wrappedResponse.toString();
       }
-      
+
       final byte[] outputBytes = output.getBytes("UTF-8");
       response.setContentLength(outputBytes.length);
       response.getOutputStream().write(outputBytes);
     }
   }
 
+  String securityContextJson(final User user) {
+    final String userJson = escapedJson(user);
+
+    return "{\"" + SecurityConstants.DICTIONARY_USER + "\": \"" + userJson + "\"}";
+  }
+
+  String escapedJson(final User user) {
+    final LookupTranslator translator = new LookupTranslator(new String[][]{{"\"", "\\\""}});
```

Comment:
It looks like you're now also protecting against use of double-quotes in the User's Roles/Groups; however should you not also then be protecting against \ too? Also, isn't a ```String.replaceAll(...)``` sufficient? Does ```LookupTranslator``` really bring anything new to the table!!?